File: internal/pkg/syncwire/accountchannel_test.go
package syncwire_test

import (
	"testing"

	"github.com/stretchr/testify/require"

	"github.com/agentre-hub/agentre/internal/pkg/syncwire"
)

func TestDecodeAccountChannelFrameProtobufCompatibility(t *testing.T) {
	// Given protobuf(AccountSyncVersion{version: 42}) and a future field 99.
	payload := []byte{0x0a, 0x04, 0x0a, 0x02, 0x08, 0x2a, 0x98, 0x06, 0x07}

	// When the public account-channel boundary decodes the frame.
	frame, known, err := syncwire.DecodeAccountChannelFrame(payload)

	// Then the known signal survives the field added by a future sender.
	require.NoError(t, err)
	require.True(t, known)
	require.Equal(t, syncwire.AccountChannelFrame{
		Type:    syncwire.AccountChannelSyncVersion,
		Version: 42,
	}, frame)
}

File: frontend/packages/agentre-wire/src/__tests__/account-channel.test.ts
import { describe, expect, it } from "vitest";

import {
  AccountChannelSyncVersion,
  ProtobufAccountChannelCodec,
} from "../index";

describe("account channel wire contract", () => {
  it("given a sync signal, when protobuf encodes it, then Go-compatible bytes survive future fields", () => {
    const payload = ProtobufAccountChannelCodec.encode({
      type: AccountChannelSyncVersion,
      version: 42,
    });

    expect(Array.from(payload)).toEqual([0x0a, 0x04, 0x0a, 0x02, 0x08, 0x2a]);

    const withFutureField = Uint8Array.from([...payload, 0x98, 0x06, 0x07]);
    expect(ProtobufAccountChannelCodec.decode(withFutureField)).toEqual({
      type: AccountChannelSyncVersion,
      version: 42,
    });
  });

  it("ignores future notifications and rejects malformed protobuf", () => {
    expect(
      ProtobufAccountChannelCodec.decode(
        Uint8Array.from([0x0a, 0x03, 0x98, 0x06, 0x01]),
      ),
    ).toBeNull();
    expect(() =>
      ProtobufAccountChannelCodec.decode(Uint8Array.from([0x0a, 0xff])),
    ).toThrow();
  });
});
